### PR TITLE
Fixed Vi mode indicator issue

### PR DESCRIPTION
**Pull request by sei40kr**
**Changes**
* Fixed an issue that Vi mode won't appear before the user changes the mode. (This change shows INSERT when failed to get current Vi mode)
* ~Added nice-exit-code plugin. This can be activated with the flag `POWERLEVEL9K_STATUS_NICE_EXIT_CODE` and shows a signal name of the last status code.~

### CODE REVIEW

**Comment by sei40kr** [comment]
This is how it works.

screenshot from 2017-07-26 23-03-40

**Comment by bhilburn** [comment]
Hey @sei40kr! Thanks so much for the bug fix and new feature!

We are pretty strict about keeping new feature merges on to the `next` branch for stability reasons on `master`. Would you mind doing one of the following?

1. Refile this entire PR against `next`
OR
2. Break this into two PRs, file the bugfix against `master`, and the new feature against `next`

I look forward to getting this merged! It looks :+1:

File: powerlevel9k.zsh-theme [hunk 1]
```diff
@@ -1006,10 +1006,16 @@ prompt_ssh() {
...
 set_default POWERLEVEL9K_STATUS_VERBOSE true
 set_default POWERLEVEL9K_STATUS_OK_IN_NON_VERBOSE false
+set_default POWERLEVEL9K_STATUS_NICE_EXIT_CODE false
 prompt_status() {
   if [[ "$RETVAL" -ne 0 ]]; then
     if [[ "$POWERLEVEL9K_STATUS_VERBOSE" == true ]]; then
-      "$1_prompt_segment" "$0_ERROR" "$2" "red" "226" "$RETVAL" 'CARRIAGE_RETURN_ICON'
+      if [[ "$POWERLEVEL9K_STATUS_NICE_EXIT_CODE" == true ]]; then
+        "$1_prompt_segment" "$0_ERROR" "$2" "red" "226" \
+          "$(nice_exit_code "$RETVAL" | awk -F ':' '{print $2 "(" $1 ")"}')" 'CARRIAGE_RETURN_ICON'
```
Comment (docwhat) [inline powerlevel9k.zsh-theme]: I don't see `nice_exit_code` anyplace.  Is something missing?

Also splitting things apart can be done without `awk`; I could fix that later if I knew what `nice_exit_code` does.

This also conflicts with another commit: 543bd2f8. `*`cough`*`  `POWERLEVEL9K_STATUS_SHOW_PIPESTATUS``

File: powerlevel9k.zsh-theme [hunk 2]
```diff
@@ -1192,11 +1198,11 @@ set_default POWERLEVEL9K_VI_INSERT_MODE_STRING "INSERT"
...
   case ${KEYMAP} in
-    main|viins)
-      "$1_prompt_segment" "$0_INSERT" "$2" "$DEFAULT_COLOR" "blue" "$POWERLEVEL9K_VI_INSERT_MODE_STRING"
-    ;;
     vicmd)
-      "$1_prompt_segment" "$0_NORMAL" "$2" "$DEFAULT_COLOR" "default" "$POWERLEVEL9K_VI_COMMAND_MODE_STRING"
+      "$1_prompt_segment" "$0_NORMAL" "$2" "$DEFAULT_COLOR" "blue" "$POWERLEVEL9K_VI_COMMAND_MODE_STRING"
+    ;;
+    main|viins|*)
+      "$1_prompt_segment" "$0_INSERT" "$2" "$DEFAULT_COLOR" "default" "$POWERLEVEL9K_VI_INSERT_MODE_STRING"
```
Comment (docwhat) [inline powerlevel9k.zsh-theme]: This should be in a different PR.

File: powerlevel9k.zsh-theme (hunk 1 above)
Comment (bhilburn) [inline powerlevel9k.zsh-theme]: Nice catches, @docwhat, on `nice_exit_code` and the conflict.

@sei40kr - Once you rebase your branch onto `next` you'll see the conflict.

**Comment by docwhat** [comment]
I stole your idea and implemented it in #581.  :heart:

**Comment by bhilburn** [comment]
@sei40kr - Your exit code idea is really great :) @docwhat has pulled it out into #581, separately.

So, we still need to get your bugfix for Vi-mode merged. You can either re-file this PR (keep it against `master`) with only 5f2e0ecb7ae, or I can cherry-pick it over onto `master`. Do you have a preference? You're commit will be in the tree either way, but having PRs is nice as an additional log, of sorts.

File: powerlevel9k.zsh-theme [hunk 3]
```diff
@@ -1192,12 +1192,12 @@ set_default POWERLEVEL9K_VI_INSERT_MODE_STRING "INSERT"
...
   case ${KEYMAP} in
-    main|viins)
-      "$1_prompt_segment" "$0_INSERT" "$2" "$DEFAULT_COLOR" "blue" "$POWERLEVEL9K_VI_INSERT_MODE_STRING"
-    ;;
     vicmd)
       "$1_prompt_segment" "$0_NORMAL" "$2" "$DEFAULT_COLOR" "default" "$POWERLEVEL9K_VI_COMMAND_MODE_STRING"
     ;;
+    main|viins|*)
+      "$1_prompt_segment" "$0_INSERT" "$2" "$DEFAULT_COLOR" "blue" "$POWERLEVEL9K_VI_INSERT_MODE_STRING"
+    ;;
```
Comment (sei40kr) [inline powerlevel9k.zsh-theme]: Added `*` as a pattern to show Vi mode correctly.
But this can show an indicator of INSERT even when the shell is not using Vi mode.

**Comment by sei40kr** [comment]
@bhilburn @docwhat 
Thank you for your review.
I've edited this PR and make this contain only a commit which fixes Vi mode issue.

**Comment by bhilburn** [comment]
@sei40kr - Excellent work! Thanks so much for incorporating the review feedback. This is a great contribution to P9k.

Welcome to the project `=)`
